src/Console/Command/FixCommand.php

```diff
@@ -227,7 +227,7 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             [
                 'allow-risky' => $input->getOption('allow-risky'),
                 'config' => $passedConfig,
-                'dry-run' => $input->getOption('dry-run'),
+                'dry-run' => $this->isDryRun($input),
```

Comment:
that means for scope of FixCommand, someone could use `isDryRun()`, but could still use `->getOption('dry-run')`. And when using in inheriting class CheckCommand, they will have different results.

Maybe instead of introducing 2nd way to get the value (custom getter), we could cal `setOption('dry-run', ...)` in child class?

Reply:
I already tried that, but since I re-create command's definition and filter out `--dry-run` option (so it's not displayed in the `--help`), it can't be set with `$input->setOption('dry-run', true)` before invoking `parent::execute($input, $output)`. It ends up with an error:

<img width="1649" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/fcee3162-b1bd-4e91-822e-14e32cb2ffb9">

We could get rid of `isDryRun()` if we don't mind displaying `--dry-run` as an `check` command's option 🙂.

Reply:
i'm not fully happy here, but the only other solution i found so far was to have yet-another-parent like `FixAbstract` and `Fix extends FixAbstract` + `Check extends FixAbstract`, but also not the best.
let's keep as-is unless anything better comes to anyone mind

Reply:
What do you mean by "keep as-is"? Merge in this form or keep PR opened until we figure something out 😅?

Reply:
ok, i see the confusion

i meant i will merge it now